### ISSUE TITLE
Windows Integration of file types gets error of invalid path

### ISSUE THREAD

**Issue by mjross** [Komodo/KomodoEdit#3183]
### Summary
When using the Preferences > Windows Integration feature for associating Komodo IDE as the default editor for certain file types (file extensions), after adding multiple extensions using the "Add" button in the "Make Komodo the default editor for these file extensions" column, when I click either the "OK" or "Apply" button, I get the message "There was an error creating or removing Windows file associations: The specified path is invalid". Unfortunately, it's not telling me what the path is or where it is being specified. The only time I specified any path was during installation, when the installer asked for the destination directory. I did not see this behavior in version 10. 

### Steps to Reproduce
Help > Troubleshooting > Restart in Safe-Mode. Edit > Preferences > Windows Integration > Add > enter the list of extensions -- included here in case it is relevant:
`.conf;.htaccess;.bsh;.css;.less;.sass;.scss;.csv;.log;.tsv;.config;.info;.install;.module;.profile;.test;.twig;.make;.patch;.inf;.ini;.js;.json;.bash;.csh;.ksh;.sh;.md;.bin;.dic;.asc;.php;.inc;.tpl;.py;.opml;.sql;.txf;.txt;.xml;.yml`

### Platform Information
Komodo IDE 11.0.0 on Windows 10 Pro

### Attachments
![file associations](https://user-images.githubusercontent.com/829492/30995057-f84ed978-a474-11e7-9ff3-8b6b5f356ea7.jpg)

[pystderr.log.txt](https://github.com/Komodo/KomodoEdit/files/1342685/pystderr.log.txt)

**Comment by cgchoffman** [comment]
Thanks for the repro steps.  Made the test easier to be thorough.  Unfortunately I can't reproduce the issue and there isn't anything in the log file your shared.  Are you sure you reproduced the issue before grabbing that log?

**Comment by mjross** [comment]
No problem. Whatever it takes to make debugging easier.

The earlier screenshot suggests I probably (and unthinkingly) changed the scheme (to Cupertino) between the two steps "Restart in Safe-Mode" and "Edit > Preferences > Windows Integration...". So just now I redid the steps fresh. Similar attachments as before. Please correct me if I'm wrong, but I'm assuming I don't have to close Komodo IDE before attaching the log file?

![file extensions error](https://user-images.githubusercontent.com/829492/31199526-e872a6d4-a914-11e7-9ef5-a711381334fa.jpg)

[pystderr.log.txt](https://github.com/Komodo/KomodoEdit/files/1357351/pystderr.log.txt)

**Comment by cgchoffman** [comment]
> Please correct me if I'm wrong, but I'm assuming I don't have to close Komodo IDE before attaching the log file?

You are not wrong :)  Still nothing in the logs.  So you reproduced this in safe mode?  You mentioned it in your last comment.

**Comment by mjross** [comment]
Yes, in safe mode.

Any ideas what sort of path it's looking for?

**Comment by cgchoffman** [comment]
It looks like that error comes from Windows when it doesn't know what registry key you're looking for.
In our case it tries to open `Software\ActiveState\Komodo IDE`.  Do you mind open regedit and seeing if you have that path?
![capture](https://user-images.githubusercontent.com/655154/31203945-41fa39c0-a91e-11e7-8849-45e825f938cd.PNG)

**Comment by mjross** [comment]
No problem. Yes, that path is present.

![registry](https://user-images.githubusercontent.com/829492/31207138-a2e3b5ac-a939-11e7-85be-c82d14d6c9ab.jpg)

**Comment by cgchoffman** [comment]
> Perhaps "There was an error creating or removing Windows file associations: The specified Windows registry key is invalid".

I agree this error isn't very helpful.  The last segment is from Windows though and I'd be reluctant to rewrite Windows errors though.

I unfortunately need to prioritize other bugs over this one, @mjross.  I can't reproduce this and I'm not aware of anyone else hitting this.  Have you tried reinstalling Komodo?

**Comment by mjross** [comment]
Regarding prioritization, I completely understand.

I haven't tried reinstalling Komodo again, because I installed a cleanly just days ago and noticed this problem when setting my preferences for the first time.

So when Komodo tries to set the registry values, and it gets an error back from Windows, does Windows provide any useful information as to what is truly blocking the changes? (I'm assuming not.)

**Comment by cgchoffman** [comment]
I wouldn't be surprised if Windows gave that error for multiple things.  As far as I can tell in this case though (i can't reproduce the issue) it's coming from the key path that I mentioned above.  When i googled the error someones command to set a value for a key in the registry and the error was a result of a bad key path.

**Comment by cgchoffman** [comment]
Oh forgot to confirm (this is more for me later) that I tested and upgraded profile that had set the same file extensions to be associated with Komodo.

**Comment by mjross** [comment]
Okay, thank you for the updates. This may be an error that is simply specific to my Windows installation, for reasons unknown.

As a workaround, I could always manually associate the more important extensions.

**Comment by mjross** [comment]
Earlier I commented that "This may be an error that is simply specific to my Windows installation, for reasons unknown." Now I have a new laptop with a pristine Windows install. So I tried the Windows Integration process again, and the message "There was an error creating or removing Windows file associations: The specified path is invalid" didn't appear.
